pkg/apis/pipeline/v1beta1/taskrun_validation.go

```diff
@@ -93,6 +95,43 @@ func (ts *TaskRunSpec) Validate(ctx context.Context) (errs *apis.FieldError) {
 	return errs
 }
 
+// validateInlineParameters validates parameters that are defined inline.
+// This is cruicial for propagated parameters since the parameters could
+// be defined under taskRun and then called directly in the task steps.
+// In this case, paratemers cannot be validated by the underlying taskSpec
+// since they may not have the parameters declared because of propagation.
+func (ts *TaskRunSpec) validateInlineParameters(ctx context.Context) (errs *apis.FieldError) {
+	errs = errs.Also(ValidateParameters(ctx, ts.Params).ViaField("params"))
+	params := ts.Params
+	var paramSpec []ParamSpec
+	if ts.Params == nil || ts.TaskSpec == nil {
```

Comment:
it seems like it would be a problem if the task spec used some propagated params, but the taskrun didn't declare them. Is this case tested?

Reply:
I don't think this comment was addressed.

Reply:
I think what I am looking for is a test case like:
```
		taskRun: &v1beta1.TaskRun{
			ObjectMeta: metav1.ObjectMeta{Name: "tr"},
			Spec: v1beta1.TaskRunSpec{
				TaskSpec: &v1beta1.TaskSpec{
					Steps: []v1beta1.Step{{
						Name:    "echo",
						Image:   "ubuntu",
						Command: []string{"echo"},
						Args:    []string{"$(params.task-words[*])"},
					}, 
				},
			},
		},
```
(this is invalid.) Does this exist anywhere?

Reply:
Thanks! it makes sense now. It would just return without throwing an error. I now added a test [here](https://github.com/tektoncd/pipeline/pull/5143/files#diff-1055ae489118d479fe9b003ef2753af20cfe8b2589bf866cb25f5228fa242b68) to capture it.

Thanks a lot!!!